Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/clibs/mime/core.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/clibs/mime/core.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/clibs/socket/core.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/clibs/socket/core.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libjpeg.8.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libjpeg.8.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwx.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwx.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwxlua-wx28d-2.8.12.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwxlua-wx28d-2.8.12.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwxlua_bind-wx28d-2.8.12.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwxlua_bind-wx28d-2.8.12.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwxlua_debug-wx28d-2.8.12.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwxlua_debug-wx28d-2.8.12.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwxlua_lua51-wx28d-2.8.12.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwxlua_lua51-wx28d-2.8.12.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwxlua_socket-wx28d-2.8.12.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwxlua_socket-wx28d-2.8.12.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/libwxstedit-wx28d-1.6.0.dylib and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/libwxstedit-wx28d-1.6.0.dylib differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/lua and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/lua differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/bin/lua.app/Contents/MacOS/lua and pkulchenko-ZeroBraneStudio-ab17270.cmake/bin/lua.app/Contents/MacOS/lua differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/build/.CMakeLists.txt.swp and pkulchenko-ZeroBraneStudio-ab17270.cmake/build/.CMakeLists.txt.swp differ
diff -urN pkulchenko-ZeroBraneStudio-ab17270/build/CMakeLists.txt pkulchenko-ZeroBraneStudio-ab17270.cmake/build/CMakeLists.txt
--- pkulchenko-ZeroBraneStudio-ab17270/build/CMakeLists.txt	1969-12-31 18:00:00.000000000 -0600
+++ pkulchenko-ZeroBraneStudio-ab17270.cmake/build/CMakeLists.txt	2012-10-16 10:02:17.742769245 -0500
@@ -0,0 +1,139 @@
+# setup the minimum allowed CMake version
+cmake_minimum_required(VERSION 2.8)
+
+# set the default build type (this needs to be done *before* project command)
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "RelWithDebInfo" "MinSizeRel")
+endif()
+
+# restrict configuration types to the selected build type
+set(CMAKE_CONFIGURATION_TYPES ${CMAKE_BUILD_TYPE} CACHE INTERNAL "" FORCE)
+
+# set the project name
+project("zbstudio")
+
+# set the top-level project directory relative to CMakeLists.txt location
+set(TOPDIR ..)
+
+# checks if the Lua module is available
+function(check_lua_module MODULE)
+  if(NOT ${MODULE}_FOUND)
+    # find Lua executable at first
+    find_program(LUA_EXECUTABLE "lua")
+    if(NOT LUA_EXECUTABLE)
+      message(FATAL_ERROR "Lua executable is not found")
+    endif()
+
+    # run a short script with "require" statement to determine if the Lua module is available
+    execute_process(COMMAND "${LUA_EXECUTABLE}" -e "require(\"${MODULE}\")"
+      RESULT_VARIABLE EXIT_CODE OUTPUT_QUIET ERROR_QUIET)
+    if(EXIT_CODE EQUAL 0)
+      set(${MODULE}_FOUND TRUE CACHE INTERNAL "")
+      message(STATUS "Found Lua module: ${MODULE}")
+    elseif(ARGV1)
+      message(FATAL_ERROR "Lua module \"${MODULE}\" is not found")
+    endif()
+  endif()
+endfunction()
+
+# adds an user option to select between system-wide and bundled Lua module
+function(add_system_lua_module_option MODULE)
+  string(TOUPPER ${MODULE} UPPERCASE_MODULE)
+  if(NOT DEFINED USE_SYSTEM_${UPPERCASE_MODULE})
+    option(USE_SYSTEM_${UPPERCASE_MODULE}
+      "Use a system-wide \"${MODULE}\" Lua module instead of the bundled one." ${${MODULE}_FOUND})
+  endif()
+endfunction()
+
+# installs all files listed in the manifest
+function(install_from_manifest TYPE MANIFEST DESTDIR EXCLUDE_REGEX)
+  file(STRINGS ${MANIFEST} FILELIST)
+  foreach(FILENAME ${FILELIST})
+    string(STRIP ${FILENAME} FILENAME)
+    if(NOT FILENAME MATCHES ${EXCLUDE_REGEX})
+      get_filename_component(FILEPATH ${FILENAME} PATH)
+      install(${TYPE} ${TOPDIR}/${FILENAME} DESTINATION ${DESTDIR}/${FILEPATH})
+    endif()
+  endforeach()
+endfunction()
+
+if(WIN32)
+  # add the executable file to our project
+  add_executable(zbstudio WIN32 ${TOPDIR}/build/win32_starter.c ${TOPDIR}/zbstudio/res/zbstudio.rc)
+
+  # link to the static multi-threaded CRT under MSVC
+  if(MSVC)
+    string(TOUPPER ${CMAKE_BUILD_TYPE} BUILD_TYPE)
+    foreach(FLAGS_VAR CMAKE_C_FLAGS CMAKE_CXX_FLAGS CMAKE_C_FLAGS_${BUILD_TYPE} CMAKE_CXX_FLAGS_${BUILD_TYPE})
+      string(REGEX REPLACE "/MD" "/MT" ${FLAGS_VAR} ${${FLAGS_VAR}})
+      string(REGEX REPLACE "/MDd" "/MTd" ${FLAGS_VAR} ${${FLAGS_VAR}})
+    endforeach()
+  endif()
+
+  # setup the data directory
+  set(DATADIR .)
+
+  # install IDE executable
+  install(TARGETS zbstudio DESTINATION ${DATADIR})
+
+  # install files from manifest
+  install_from_manifest(FILES ${TOPDIR}/zbstudio/MANIFEST ${DATADIR} "^$")
+  install_from_manifest(FILES ${TOPDIR}/zbstudio/MANIFEST-bin-win32 ${DATADIR} "^zbstudio.exe$")
+elseif(APPLE)
+  # setup the data directory
+  set(ROOTDIR ZeroBraneStudio.app/Contents)
+  set(DATADIR ${ROOTDIR}/ZeroBraneStudio)
+
+  # install IDE shell script
+  install(PROGRAMS ${TOPDIR}/zbstudio/ZeroBraneStudio.app/Contents/MacOS/ZeroBraneStudio DESTINATION ${ROOTDIR}/MacOS)
+
+  # install icon pack and .plist file
+  install(DIRECTORY ${TOPDIR}/zbstudio/ZeroBraneStudio.app/Contents/Resources DESTINATION ${ROOTDIR})
+  install(FILES ${TOPDIR}/zbstudio/ZeroBraneStudio.app/Contents/Info.plist DESTINATION ${ROOTDIR})
+
+  # install files from manifest
+  install_from_manifest(FILES ${TOPDIR}/zbstudio/MANIFEST ${DATADIR} "^$")
+  install_from_manifest(PROGRAMS ${TOPDIR}/zbstudio/MANIFEST-bin-macos ${DATADIR} "^$")
+else()
+  # check the available Lua modules
+  check_lua_module(wx TRUE)
+  check_lua_module(socket TRUE)
+  check_lua_module(copas ${USE_SYSTEM_COPAS})
+  check_lua_module(luainspect ${USE_SYSTEM_LUAINSPECT})
+  #check_lua_module(mobdebug ${USE_SYSTEM_MOBDEBUG})
+
+  # add user options to select between system-wide and bundled Lua modules
+  add_system_lua_module_option(copas)
+  add_system_lua_module_option(luainspect)
+  #add_system_lua_module_option(mobdebug)
+
+  # setup the data directory
+  set(DATADIR share/zbstudio)
+
+  # install IDE shell script
+  set(IDE_DATADIR "${CMAKE_INSTALL_PREFIX}/${DATADIR}")
+  configure_file(${TOPDIR}/zbstudio/zbstudio.in "${CMAKE_BINARY_DIR}/zbstudio")
+  install(PROGRAMS "${CMAKE_BINARY_DIR}/zbstudio" DESTINATION bin)
+
+  # install bundled Lua modules
+  if(NOT USE_SYSTEM_COPAS)
+    install(DIRECTORY ${TOPDIR}/lualibs/copas ${TOPDIR}/lualibs/coxpcall DESTINATION ${DATADIR}/lualibs)
+  endif()
+  if(NOT USE_SYSTEM_LUAINSPECT)
+    install(DIRECTORY ${TOPDIR}/lualibs/luainspect ${TOPDIR}/lualibs/metalua DESTINATION ${DATADIR}/lualibs)
+  endif()
+  if(NOT USE_SYSTEM_MOBDEBUG)
+    install(DIRECTORY ${TOPDIR}/lualibs/mobdebug DESTINATION ${DATADIR}/lualibs)
+  endif()
+
+  # install .desktop file and hicolor icon theme
+  install(DIRECTORY ${TOPDIR}/zbstudio/res/icons/ DESTINATION share/icons/hicolor)
+  install(FILES ${TOPDIR}/zbstudio/res/zbstudio.desktop DESTINATION share/applications)
+
+  # install miscellaneous documentation files
+  install(FILES ${TOPDIR}/CHANGELOG.md ${TOPDIR}/LICENSE ${TOPDIR}/README.md DESTINATION share/doc/zbstudio)
+
+  # install files from manifest
+  install_from_manifest(FILES ${TOPDIR}/zbstudio/MANIFEST ${DATADIR} "^(CHANGELOG.md|LICENSE|README.md)$|^lualibs/")
+endif()
Binary files pkulchenko-ZeroBraneStudio-ab17270/build/ZeroBraneStudio.dmg.bz2 and pkulchenko-ZeroBraneStudio-ab17270.cmake/build/ZeroBraneStudio.dmg.bz2 differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/build/wxlua28_2.8.12.2-1_i386.deb and pkulchenko-ZeroBraneStudio-ab17270.cmake/build/wxlua28_2.8.12.2-1_i386.deb differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/ZeroBraneStudio.app/Contents/Resources/zbstudio.icns and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/ZeroBraneStudio.app/Contents/Resources/zbstudio.icns differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_BREAK.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_BREAK.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_BREAKPOINT_TOGGLE.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_BREAKPOINT_TOGGLE.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_CALLSTACK.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_CALLSTACK.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_START.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_START.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_STEP_INTO.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_STEP_INTO.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_STEP_OUT.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_STEP_OUT.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_STEP_OVER.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_STEP_OVER.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_STOP.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_STOP.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/16/wxART_DEBUG_WATCH.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/16/wxART_DEBUG_WATCH.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/16x16/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/16x16/apps/zbstudio.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/22x22/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/22x22/apps/zbstudio.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/24x24/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/24x24/apps/zbstudio.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/256x256/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/256x256/apps/zbstudio.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/32x32/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/32x32/apps/zbstudio.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/48x48/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/48x48/apps/zbstudio.png differ
Binary files pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/icons/64x64/apps/zbstudio.png and pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/icons/64x64/apps/zbstudio.png differ
diff -urN pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/zbstudio.desktop pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/zbstudio.desktop
--- pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/zbstudio.desktop	1969-12-31 18:00:00.000000000 -0600
+++ pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/zbstudio.desktop	2012-10-16 10:02:18.089434246 -0500
@@ -0,0 +1,13 @@
+[Desktop Entry]
+Type=Application
+Version=1.0
+Encoding=UTF-8
+Name=ZeroBrane Studio
+GenericName=Integrated Development Environment
+Comment=A simple and extensible Lua IDE and debugger
+Exec=zbstudio %F
+Icon=zbstudio
+Terminal=false
+Categories=GTK;Development;IDE;
+MimeType=text/plain;application/lua;
+StartupNotify=true
diff -urN pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/zbstudio.rc pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/zbstudio.rc
--- pkulchenko-ZeroBraneStudio-ab17270/zbstudio/res/zbstudio.rc	1969-12-31 18:00:00.000000000 -0600
+++ pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/res/zbstudio.rc	2012-10-16 10:02:18.089434246 -0500
@@ -0,0 +1 @@
+icon ICON "32.ico"
diff -urN pkulchenko-ZeroBraneStudio-ab17270/zbstudio/zbstudio.in pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/zbstudio.in
--- pkulchenko-ZeroBraneStudio-ab17270/zbstudio/zbstudio.in	1969-12-31 18:00:00.000000000 -0600
+++ pkulchenko-ZeroBraneStudio-ab17270.cmake/zbstudio/zbstudio.in	2012-10-16 10:02:18.092767564 -0500
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+cd "@IDE_DATADIR@"
+lua src/main.lua zbstudio
